packages/opentelemetry-node/src/spanprocessor.ts

```diff
@@ -0,0 +1,115 @@
+import { Context } from '@opentelemetry/api';
+import { Span as OtelSpan, SpanProcessor as OtelSpanProcessor } from '@opentelemetry/sdk-trace-base';
+import { getCurrentHub } from '@sentry/core';
+import { Span as SentrySpan, TransactionContext } from '@sentry/types';
+import { logger } from '@sentry/utils';
+
+/**
+ * Converts OpenTelemetry Spans to Sentry Spans and sends them to Sentry via
+ * the Sentry SDK.
+ */
+export class SentrySpanProcessor implements OtelSpanProcessor {
+  // public only for testing
+  public readonly _map: Map<SentrySpan['spanId'], SentrySpan> = new Map<SentrySpan['spanId'], SentrySpan>();
```

Comment:
It's not _super_ nice to make this public just so we can use it in tests, so if there are other ideas for this, let me know. It works, at least 😅 

Also FYI I changed this to a `Map` instead of `{}`, as that makes `delete()` nicer imho. I _think_ there shouldn't really be a perf difference, but let me know if we should rather leave this a POJO.

Reply:
Using a map seems fine to me - cleaner pattern overall.